### ISSUE TITLE
mockDOMSource: DOM.select(...).select is not a function

### ISSUE THREAD

**Issue by mspoulsen** [cyclejs/core#265]
Greeting from the numero uno cycle.js fan! :)

I have a question and it may very well be me who is doing something wrong. I am using mockDOMSource to test my components and I run into trouble when I tag on multiple selects like this:

sources.DOM
                .select('.tool-bar-build')
                .select('.new-game')
                .events('click');

I get: Uncaught TypeError: sources.DOM.select(...).select is not a function

However, the code works fine with the real DOM, so my question is if mockDOMSource does not yet support multiple selects like this?

Is there an easy workaround?

Thanks!

**Comment by staltz** [comment]
Thanks for finding this. I think it's actually a legit bug rather than a feature suggestion. mockDOMSource should make a DOMSource that has the same API as the real one.

**Comment by Cmdv** [comment]
@staltz will this be simple as making `mockDOMSource` curried? having a quick peep

**Comment by staltz** [comment]
Yes @Cmdv a PR would be great! Remember to add a test for it. At this point (we are rewriting), tests are very important.

**Comment by Cmdv** [comment]
just working on PR, got test but trying to work out currying with dot notation `.select(...).select(...)` I'll send PR once I work that out.

**Comment by staltz** [comment]
Fixed in Cycle DOM v9.2.1 thanks to @Cmdv

**Comment by mspoulsen** [comment]
Hi,

Thanks for the effort. It now works as long as I make sure I pass in all the events the component is listening to, but that should be necessary right? That would get very brittle with nested components very quickly. 

Let me illustrate:

```javascript
function component ({ DOM }) {

    const click$   = DOM
                        .select('.foo')
                        // if I comment out this 2nd select it works
                        // if not I get same error
                        .select('.bar')      
                        .events('click');
    return {
        click$
    }
}

const click$ = component({ DOM : mockDOMSource({}) }).click$;

click$
    .do(x => console.log(x))
    .subscribe();
```

**Comment by staltz** [comment]
@Cmdv can you check this? I think we need to add tests for those cases where people give `{}` in `mockDOMSource({})`

**Comment by Cmdv** [comment]
yeah cool I'll pick it up tonight @staltz

**Comment by staltz** [comment]
Released `@cycle/dom` v9.2.2 with a fix, thanks to @Cmdv
